### PR TITLE
Replaced Array.map+slice+reduce+filter with Array.some

### PR DESCRIPTION
**Pull request by regiskuckaertz**
A small change which could potentially yield great results (think cut-and-paste large documents). In `includeRealMutations`, we just want to return as soon as a matching node is found, hence `Array.some`.

### CODE REVIEW

File: src/dom-observer.js [hunk 1]
```diff
@@ -3,24 +3,23 @@ define([
...
+    window.MozMutationObserver;
+
+  function hasRealMutation(n) {
+    return ! nodeHelpers.isEmptyTextNode(n) &&
+      ! elementHelpers.isSelectionMarkerNode(n);
+  }
 
+  function includeRealMutations(mutations) {
+    return mutations.some(function(mutation) {
+      return mutations.some.call(mutation.addedNodes, hasRealMutation) ||
```
Comment (rrees) [inline src/dom-observer.js]: Should the inner some actually be a prototype call since you're passing in the new data set to evaluate?

Comment (regiskuckaertz) [inline src/dom-observer.js]: Yes, in this case `mutation**s**.some` is `Array.prototype.some`, it just saved me a few keystrokes :-)

Comment (rrees) [inline src/dom-observer.js]: Can you put the full version to make it easier to read/maintain? It threw me at first.

Comment (regiskuckaertz) [inline src/dom-observer.js]: Of course! Done.

**Comment by rrees** [comment]
:+1:

**Pull request merged**